openverse_catalog/dags/providers/provider_workflows.py

```diff
@@ -26,6 +27,47 @@
     WikimediaCommonsDataIngester,
 )
 from providers.provider_api_scripts.wordpress import WordPressDataIngester
+from typing_extensions import NotRequired, TypedDict
+
+
+class TaskOverride(TypedDict):
+    """
+    Describes available options for overriding a task's properties.
+
+    Required Arguments:
+
+    task_id: A regex pattern that matches the task_id of the task
+             to be modified.
+
+    Optional Arguments:
+
+    timeout: str in "%d:%H:%M:%S" format giving the amount of time the
+             task may take
+    """
+
+    task_id_pattern: str
+    timeout: NotRequired[str | None]
+
+
+def get_time_override(time_str: str | None) -> timedelta | None:
+    """
+    Convert a string in the format "%d:%H:%M:%S" to a timedelta.
+    Return None if the string is improperly formatted.
+
+    Example: "5:10:20:30" represents 5 days, 10 hours, 20 minutes, and
+    30 seconds.
+    """
+    if time_str is None:
+        return None
+
+    ts = time_str.split(":")
+    if len(ts) != 4 or not all(t.isdigit() for t in ts):
+        # Incorrectly formatted time string. Do not apply. We don't raise
+        # an error in order to avoid breaking DAGs.
+        return None
+
+    [days, hours, minutes, seconds] = [int(t) for t in ts]
+    return timedelta(days=days, hours=hours, minutes=minutes, seconds=seconds)
```

Comment:
I know that this will add complexity to the parsing but...do you think we could do `<days>d:<hours>h:<minutes>m:<seconds>s`? E.g. `4d:5h:0m:10s`? This might make the timeouts a little bit clearer, I was confused initially as to what the values represented.

Airflow uses [pendulum](https://pendulum.eustace.io/docs/) under the hood, so I think we could simplify this too by using something like `from_format` or similar? It doesn't look like there's a method for getting from strings to durations but we might be able to go from a datetime to a timedelta that way. What do you think?

Reply:
Oh sorry, I didn't mean to imply that we should have something more complex than that! I think your implementation looks great ✨

Reply:
Oh totally, I'm stopping _myself_ from getting carried away adding more 😄 I was initially trying to make a single regex for matching but updated it so you don't have to include all four parts 😆 I think that ought to cover realistic use cases.